android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/screens/ReportTypeSelectorScreen.kt

```diff
@@ -77,26 +74,12 @@ fun ReportTypeSelectorScreen(
   val uiState = measureReportViewModel.reportTypeSelectorUiState.value
 
   ReportTypeSelectorPage(
-    startDate = uiState.startDate.ifEmpty { stringResource(id = R.string.start_date) },
-    endDate = uiState.endDate.ifEmpty { stringResource(id = R.string.end_date) },
-    patientName = uiState.patientViewData?.name,
-    generateReport =
-      uiState.startDate.isNotEmpty() &&
-        uiState.endDate.isNotEmpty() &&
-        (uiState.patientViewData != null ||
-          measureReportViewModel.reportTypeState.value == MeasureReport.MeasureReportType.SUMMARY),
-    onGenerateReportClicked = {
-      measureReportViewModel.onEvent(MeasureReportEvent.GenerateReport(navController, context))
-    },
-    reportTypeState = measureReportViewModel.reportTypeState,
-    onReportTypeSelected = {
-      measureReportViewModel.onEvent(MeasureReportEvent.OnReportTypeChanged(it, navController))
-    },
-    showProgressIndicator = uiState.showProgressIndicator,
     screenTitle = screenTitle,
-    dateRange = measureReportViewModel.dateRange,
-    onDateRangeSelected = { newDateRange ->
-      measureReportViewModel.onEvent(MeasureReportEvent.OnDateRangeSelected(newDateRange))
+    onGenerateReportClicked = { date ->
+      measureReportViewModel.onEvent(
+        MeasureReportEvent.GenerateReport(navController, context),
+        date
+      )
     },
```

Comment:
I do not think we needed to refactor the existing measure reporting implementation. In app reporting should be implemented and configured independently. Should have it's own UI, data models, repository and viewmodel.

Reply:
I have refactored the code as we have to reuse some code from existing viewmodel. 
 @f-odhiambo  Please let me know if need to keep previous ui as it is or not ?